lib/lan_connection.js

```diff
@@ -130,21 +130,29 @@ Scanner.prototype.start = function() {
       try {
         // Start discovering Tessels
         self.browser.discover();
-      } catch (error) {
-        self.emit('error', error);
+      } catch (err) {
+        process.nextTick(function() {
+          self.emit('error', err);
+        });
       }
     });
 
     // When the browser finds a new device
     self.browser.on('update', function(data) {
-      // Check if it's a Tessel
-      if (_.findIndex(self.discovered, data) > -1) {
-        return;
+      try {
+        // Check if it's a Tessel
+        if (_.findIndex(self.discovered, data) > -1) {
+          return;
+        }
+
+        var connection = new LANConnection(data);
+
+        self.emit('connection', connection);
+      } catch (err) {
+        process.nextTick(function() {
+          self.emit('error', err);
+        });
```

Comment:
Right, I'm familiar with the synchronous nature of emit and the EventEmitter class as a whole. Looking at the bug filed [here](https://github.com/kmpm/node-mdns-js/issues/40), and a larger portion of `lib/lan_connection.js` source [here](https://github.com/LinusU/t2-cli/blob/mdns-swallow-workaround/lib/lan_connection.js#L125-L157) the issue becomes clear. Thanks :)